modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2022-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.kinesis
+
+import scala.concurrent.duration._
+
+import cats.effect.{Blocker, IO}
+import cats.effect.concurrent.Ref
+
+import fs2.Stream
+
+import retry.syntax.all._
+import retry.RetryPolicies
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.BeforeAfterAll
+
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Event
+
+import com.snowplowanalytics.snowplow.enrich.common.fs2.config.io.{Input, Output}
+
+class EnrichKinesisSpec extends Specification with BeforeAfterAll with CatsIO {
+
+  val retryPolicy =
+    RetryPolicies.limitRetriesByCumulativeDelay(5.minutes, RetryPolicies.capDelay[IO](2.second, RetryPolicies.fullJitter[IO](1.second)))
+
+  def beforeAll: Unit = Containers.startAll()
+
+  def afterAll: Unit = Containers.stopAll()
+
+  "enrich-kinesis" should {
+    "initialize with the minimal config" in {
+      val isRunning = IO(Containers.enrichMinimal.isRunning())
+      val result =
+        for {
+          _ <- isRunning.retryingOnFailures(
+            _ == false,
```

Comment:
I dropped the `WaitStrategy` for Enrich containers because it turns out that we don't need it.

For `stop` I think that we can't do without the retry with timeout, because we especially want to check that the container stops and doesn't get stuck, whatever the logs contain.